rlkit/core/rl_algorithm.py

```diff
@@ -482,6 +495,128 @@ def get_extra_data_to_save(self, epoch):
             data_to_save['algorithm'] = self
         return data_to_save
 
+
+    ##### Eval stuff #####
+    def obtain_eval_paths(self, deterministic=False):
+        '''
+        collect paths with current policy
+        each transition will update the running latent context
+        the context used to condition the policy can be resampled at different intervals
+        (to enable trajectory-level or transition-level adaptation, for example)
+        '''
+        self.reset_posterior()
+        test_paths = self.eval_sampler.obtain_samples(deterministic=deterministic, resample=self.resample_z)
```

Comment:
does this ever update the posterior over z?

Reply:
yes, during eval every transition is added to the context, and the `resample` arg controls how often a new `z` is sampled from this updated posterior.

Reply:
this does seem a little opaque, as it's not obvious that at test time the agent updates the posterior.

Reply:
yea I agree...see my comment on the `obtain_samples()` function...if we don't care about transition-level updating, we can make it more clear.